backend/inspirehep/serializers.py

```diff
@@ -321,3 +322,59 @@ def dumps(obj, app=None, **kwargs):
             return rv.encode(encoding)
 
     return rv
+
+
+def record_responsify(serializer, mimetype):
+    """Create a Records-REST response serializer.
+    :param serializer: Serializer instance.
+    :param mimetype: MIME type of response.
+    :returns: Function that generates a record HTTP response.
+    """
+
+    def view(pid, record, code=200, headers=None, links_factory=None):
+        response = current_app.response_class(
+            serializer.serialize(pid, record, links_factory=links_factory),
+            mimetype=mimetype,
+        )
+        response.status_code = code
+        set_headers_for_record_caching_and_concurrency(response, record)
+
+        if headers is not None:
+            response.headers.extend(headers)
+
+        return response
+
+    return view
+
+
+def search_responsify(serializer, mimetype):
```

Comment:
can we add a param to be able to include links in the headers if we need to?

Reply:
we can